pkg/agent/execbridge.go

```diff
@@ -51,6 +51,25 @@ func (iface *execPluginInterface) Request(
 		iface.runner.recordResourceChange(*lastPermit, resp.Permit, iface.runner.global.metrics.schedulerApprovedChange)
 	}
 
+	successful := func() bool {
+		if err != nil { // request is failed
+			return false
+		}
+		if resp.Permit == target { // request is fully approved by the scheduler
+			return true
+		}
+		if lastPermit != nil && *lastPermit != resp.Permit { // request is partially approved by the scheduler
+			return true
+		}
+		return false // scheduler denied the request
```

Comment:
Why is partial approval a "success" but full approval is not? is it something along the lines of "partial approval amounts are monotonically increasing, so it still tracks liveness but has fewer false-positives" ?

Reply:
Full approval is also a success, no?

Reply:
The if statement above cover the full approval case, which defines it as a success.